Expand explanation of uses for dynamic blocks (#16228)
WordPress/gutenberg: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md

# Creating dynamic blocks

Dynamic blocks are blocks that build their structure and content on the fly when the block is rendered on the front end.

There are two primary uses for dynamic blocks:

> 1. Blocks where content should change even if a post has not been updated. One example from WordPress itself is the latest posts block. This block will update everywhere it is used when a new post is published.

Reviewer: ```suggestion

1. Blocks where content should change even if a post has not been updated. One example from WordPress itself is the Latest Posts block. This block will update everywhere it is used when a new post is published.
```

> 2. Blocks where updates to the code (HTML, CSS, JS) should be immediately shown on the front end of the website. For example, if you update the HTML structure of a block by adding a new class, adding a div, or changing the layout in any other way, if you want these changes to be applied immediately on all occurrences of that block across the site, a dynamic block should be used. (If a dynamic block is not used then when block code is updated Guterberg's [validation process](https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#validation) applies, causing users to see the validation message, "This block appears to have been modified externally").

Reviewer: Would recommend not including the note around validation. This is a common occurrence but there are ways around it with block transforms (which should be better documented, admittedly).

```suggestion
2. Blocks where updates to the code (HTML, CSS, JS) should be immediately shown on the front end of the website. For example, if you update the structure of a block by adding a new class, adding an HTML element, or changing the layout in any other way, using a dynamic block ensures those changes are live across your entire site at the same time.
```

> The general principle of dynamic blocks is that the built in `save` function should be returned as `null` so that Guternberg does not save the block HTML into the database. Instead Gutenberg will save only the [block attributes](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/) when the page is updated.  These attributes are then available to be used by the server-side rendering for displaying the block on the front end of your site.  (Alternatively, the return value of `save` could represent a cached copy of the block’s content to be shown only in case the plugin implementing the block is ever disabled.)

Reviewer: ```suggestion
For many dynamic blocks, the `save` callback function should be returned as `null`, which tells the editor to save only the [block attributes](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/) to the database.  These attributes are then passed into the server-side rendering callback, so you can decide how to display the block on the front end of your site.

You can also save an HTML representation of the block. If you provide a server-side rendering callback, this HTML will be replaced with the output of your callback, but will be rendered if your block is deactivated or your render callback is removed.
```

Block attributes can be used for any content or setting you want to save for that block. In the first example above, with the latest posts block, the number of latest posts you want to show could be saved as an attribute. Or in the second example, attributes can be used for each piece of content you want to show in the front end - such as heading text, paragraph text, an image, a URL, etc.

> 

Reviewer: ```suggestion
```

# Creating dynamic blocks

Dynamic blocks are blocks that build their structure and content on the fly when the block is rendered on the front end.

> There are two primary uses for dynamic blocks:

Reviewer: ```suggestion
There are two primary uses for dynamic blocks:

```
